packages/compartment-mapper/test/main.test.js

```diff
@@ -64,18 +62,21 @@ const builtinLocation = new URL(
 
 let modules;
 
-test("create builtin", async t => {
+async function setup() {
+  if (modules !== undefined) {
+    return;
+  }
   const utility = await loadLocation(read, builtinLocation);
   const { namespace } = await utility.import({ globals });
   // We pass the builtin module into the module map.
   modules = {
     builtin: namespace
   };
-  t.end();
-});
+}
```

Comment:
Since this is a differential review, it is fine. But I'd like to better understand what this is about.

Reply:
There’s an apparent difference in behavior between ava and tape. This routine is a preamble for subsequent tests that had to become more explicitly so. The `modules` object is built here and cached for subsequent tests. Tape would always run this once up front as a test. With Ava, there are no guarantees about whether tests get executed in the same process and whether they’re ordered. So, calling `setup` explicitly in each test and memoizing the result is now necessary.